playground/backend-integration/__tests__/backend-integration.spec.ts

```diff
@@ -71,12 +72,13 @@ describe.runIf(isServe)('serve', () => {
 
   test('CSS dependencies are tracked for HMR', async () => {
     const el = await page.$('h1')
-    browserLogs.length = 0
-
-    editFile('frontend/entrypoints/main.ts', (code) =>
-      code.replace('text-black', 'text-[rgb(204,0,0)]')
+    await untilBrowserLogAfter(
+      () =>
+        editFile('frontend/entrypoints/main.ts', (code) =>
+          code.replace('text-black', 'text-[rgb(204,0,0)]')
+        ),
+      '[vite] css hot updated: /global.css'
     )
     await untilUpdated(() => getColor(el), 'rgb(204, 0, 0)')
-    expect(browserLogs).toContain('[vite] css hot updated: /global.css')
```

Comment:
This previous code assumes `[vite] css` log to be output before `el`'s color was changed. But the order could be

1. color changes
2. `expect(browserLogs)` line runs
3. `[vite] css` log output